addons/pos_razorpay/static/src/js/payment_razorpay.js

```diff
@@ -0,0 +1,180 @@
+/** @odoo-module */
+
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+
+const REQUEST_TIMEOUT = 5000;
+
+export class PaymentRazorpay extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.p2pRequestId = {};
+    }
+
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._process_razorpay(cid);
+    }
+
+    pending_razorpay_line() {
+        return this.pos.getPendingPaymentLine("razorpay");
+    }
+
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._razorpay_cancel();
+    }
+
+    _razorpay_cancel() {
+        const razorpayCancelPayment = async () => {
+            const paymentLine = this.pos.get_order()?.selected_paymentline;
+            try {
```

Comment:
In this case you need to avoid try catch and use promise.
If an error occur from the DB you raise the error to send it to the cashier.
Check it with pos_viva_wallet module to compare